azurerm/resource_arm_hdinsight_test.go

```diff
@@ -0,0 +1,171 @@
+package azurerm
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccAzureRMHDInsightCluster_basic(t *testing.T) {
+	ri := acctest.RandInt()
+	config := testAccAzureRMHDInsightCluster_basic(ri, testLocation())
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMHDInsightClusterDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMHDInsightClusterExists("azurerm_hdinsight_cluster.test"),
+				),
+			},
+		},
+	})
+}
+
+func testAccAzureRMHDInsightCluster_basic(rInt int, location string) string {
+	return fmt.Sprintf(`
+		resource "azurerm_resource_group" "test" {
+			name     = "hdinsight-cluster-%d"
```

Comment:
we've got sweepers that automatically remove any left-over test resources matching the prefix `acctest` - would we be able to make this `acctestrg-%d`?